src/02-room-details.js

```diff
@@ -25,7 +25,24 @@ const exampleRoomData = require("../data/rooms");
  *  getRoomByDinosaurName(dinosaurs, rooms, "Pterodactyl");
  *  //> "Dinosaur with name 'Pterodactyl' cannot be found."
  */
-function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
+function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {
+  let results = ""
+  let roomHelp = []
+ for (let i = 0; i < dinosaurs.length;i++){
+   for (let j = 0; j < rooms.length; j++){
+   if (dinosaurName !== dinosaurs[i].name){
+     results = `Dinosaur with name '${dinosaurName}' cannot be found.`
```

Comment:
you want to be careful where you reassign results. If your dinosaur name is in the middle of the dino array and result is the dinosaur ID, when it moves on to the next dinosaur element it is going to reassign result to this string.